api/machine/machine.proto

```diff
@@ -713,3 +715,23 @@ message EtcdMemberList {
   repeated string members = 2;
 }
 message EtcdMemberListResponse { repeated EtcdMemberList messages = 1; }
+
+// rpc generateConfiguration
+// GenerateConfigurationRequest describes a request to generate a new configuration
+// on a node.
+message GenerateConfigurationRequest {
+  int32 machine_type = 1;
+  string version = 2;
```

Comment:
my suggestion is to use embedded `message` for each part of the config (cluster, machine, machine network) to make it more readable and easier to extend. we could align that closer with config structure?

Reply:
and it will actually be nested in that case. Is that fine?

Reply:
I'm not sure that matching it with actual config structure is a good option... Here we just specify generator input parameters for the generator, while internal config struct may change.
It actually becomes harder to use this gRPC having nested structs, especially now.
But we can group params like you suggested, just avoiding nesting them. What do you think?